clevertap-core/src/main/java/com/clevertap/android/sdk/ProfileValueHandler.kt

```diff
@@ -0,0 +1,221 @@
+package com.clevertap.android.sdk
+
+import com.clevertap.android.sdk.ProfileValueHandler.NumberValueType.DOUBLE_NUMBER
+import com.clevertap.android.sdk.ProfileValueHandler.NumberValueType.FLOAT_NUMBER
+import com.clevertap.android.sdk.ProfileValueHandler.NumberValueType.INT_NUMBER
+import com.clevertap.android.sdk.utils.CTJsonConverter
+import com.clevertap.android.sdk.validation.ValidationResult
+import com.clevertap.android.sdk.validation.ValidationResultFactory
+import com.clevertap.android.sdk.validation.ValidationResultStack
+import com.clevertap.android.sdk.validation.Validator
+import org.json.JSONArray
+
+class ProfileValueHandler(
+    private val validator: Validator,
+    private val validationResultStack: ValidationResultStack
+) {
+
+    internal enum class NumberValueType {
+        INT_NUMBER,
+        FLOAT_NUMBER,
+        DOUBLE_NUMBER
+    }
+
+    private var numberValueType: NumberValueType? = null
+    fun handleIncrementDecrementValues(value: Number, command: String, existingValue: Number?): Number? {
```

Comment:
can we handle nullables here in better way?

Reply:
Will check if refactoring is needed for the older logic